test/function/samples/simplified-nested-if/main.js

```diff
@@ -0,0 +1,22 @@
+if (unknown)
+	for (var x = 1; x--; x > 0)
+		if (!unknown) console.log('effect');
+		else 'Side effect free code to be dropped';
+else throw new Error("Shouldn't be reached");
+
+if (unknown)
+	for (var x = 1; x--; x > 0)
+		if ((console.log(), false)) console.log('effect');
+		else 'Side effect free code to be dropped';
+else throw new Error("Shouldn't be reached");
+
+if (unknown) {
+	for (var x = 1; x--; x > 0)
+		if (!unknown) console.log('effect');
+		else 'Side effect free code to be dropped';
+} else throw new Error("Shouldn't be reached");
+
+if (unknown)
+	for (var x = 1; x--; x > 0)
```

Comment:
Not that it matters, but the `x > 0` expressions in the 4 loops are superfluous. Perhaps you intended to swap the for loops' test with update.

Reply:
Makes sense